test/test_sync.cpp

```diff
@@ -348,35 +356,105 @@ TEST(SyncEvent, WaitForTwoNotifyOne)
 {
     Mutex mutex;
     Condition cond;
+    vector<int> notified_clients;
     cond.init();
     const steady_clock::duration timeout = seconds_from(3);
 
-    auto wait_async = [](Condition* cond, Mutex* mutex, const steady_clock::duration& timeout) {
+    volatile bool resource_ready = true;
+
+    auto wait_async = [&notified_clients, &resource_ready](Condition* cond, Mutex* mutex, const steady_clock::duration& timeout, int id) {
         UniqueLock lock(*mutex);
-        return cond->wait_for(lock, timeout);
+        if (cond->wait_for(lock, timeout) && resource_ready)
+        {
+            notified_clients.push_back(id);
+            resource_ready = false;
+            return 42;
```

Comment:
Return `true` or `1`.

```suggestion
            return true;
```